### PR TITLE
Brian hw order

### PR DESCRIPTION
**Pull request by brngit-code101**
This branch makes some changes to the order route and the mqtt client plugin. 
It will create a dictionary with the information the hw team needs and turns into a json object by the mqtt client. 
Format of the dictionary is as follows:
{ item_id: { quantity: item_quantity, weight: item_weight, volume: item_volume, row: item.row, column: item_column } }

Resolves: #40

### CODE REVIEW

File: routes/order/index.js [hunk 1]
```diff
@@ -78,7 +81,15 @@ module.exports = async function (fastify, opts) {
           }
         } else if (orderedItems[item] < stock[item]) {
           stock[item] = stock[item] - orderedItems[item]
+          itemInfo = getItemInfo(item, this.dynamo)
+          if (!itemInfo) {
+            return reply.code(400).send({
+              reason: 'item_id could not be found'
```
Comment (love-lena) [inline routes/order/index.js:87]: add missing_item_id

**Review by love-lena — CHANGES_REQUESTED** [review]

**Comment by brngit-code101** [comment]
Think I am finished with this issue. This is what the JSON sent to the hw team looks like/
{'orderID': '889331ea-05a8-4b2f-9c24-768c3eb095c8', 'orderList': {'d016': {'quantity': 2, 'weight': 2, 'volume': 3, 'row': 3, 'column': 2}, 't002': {'quantity': 3, 'weight': 0, 'volume': 0, 'row': 2, 'column': 1}}}

File: plugins/fastify-mclient.js [hunk 2]
```diff
@@ -15,9 +15,9 @@ class CustomMqttClient {
     this.mqttClient.on('message', callback)
   }
 
-  async submitOrder (orderID, machineID, orderList) {
+  async submitOrder (orderID, machineID, machineOrder) {
     const topic = machineID + '/order/vend'
-    const order = JSON.stringify({'orderID': orderID, 'orderList': orderList})
```
Comment (love-lena) [inline plugins/fastify-mclient.js:20]: opinionated nit: orderList is a better variable name for this

File: routes/order/util.js [hunk 3]
```diff
@@ -73,6 +73,33 @@ module.exports = {
...
+    const weight = itemCheckResponse.Item.weight
+    const volume = itemCheckResponse.Item.volume
+
+    const itemInfo = { 'itemWeight': weight, 'itemVolume': volume }
+    return itemInfo
+  },
+
+  createMachineOrder (machineOrder, itemID, itemQuantity, itemInfo, itemLocation) {
+    machineOrder[itemID] = {quantity: itemQuantity, weight: itemInfo['itemWeight'], volume: itemInfo['itemVolume'], row: itemLocation.row, column: itemLocation.column}
+    return machineOrder
```
Comment (love-lena) [inline routes/order/util.js:102]: Potential issue here. If an item doesn't exist, `getItemInfo` returns null. That is fine, but in createMachineOrder, you do `itemInfo['itemWeight']`. If itemInfo is null, this will throw an uncaught error because the field doesn't exist in null. You should do a null check somewhere to make sure you don't try to access a field of a null value.

Please add a unit test for `createMachineOrder` that takes in a null itemInfo, so you can verify this fix

**Review by love-lena — CHANGES_REQUESTED** [review]
1 nit pick, but one actual error you need to resolve and add test coverage for

**Review by love-lena — APPROVED** [review]
